dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/transport/AbstractH2TransportListener.java

```diff
@@ -63,7 +67,11 @@ protected Map<String, Object> headersToMap(Http2Headers trailers, Supplier<Objec
                     LOGGER.error(PROTOCOL_FAILED_PARSE, "", "", "Failed to parse response attachment key=" + key, e);
                 }
             } else {
-                attachments.put(key, header.getValue().toString());
+                try {
+                    attachments.put(key, URLDecoder.decode(header.getValue().toString(), String.valueOf(StandardCharsets.UTF_8)));
+                } catch (UnsupportedEncodingException e) {
+                    LOGGER.error(PROTOCOL_FAILED_DECODE, "", "", "Failed decode attachment key=" + key, e);
+                }
```

Comment:
Check if such is decoded by Dubbo or this may break some user's usage if they want to pass a exact value which is URLEncoded.

Reply:
> Check if such is decoded by Dubbo or this may break some user's usage if they want to pass a exact value which is URLEncoded.

I understand that this situation may due to different Dubbo versions of the consumer and provider 

If both the consumer and provider include this pr, when a user wants to pass a URL-encoded value in the consumer, Dubbo will encode it first, and in the provider dubbo will decode it, the user receives the URL-encoded value. This is as expected

However, problems may arise when different dubbo versions are used:

1、If the consumer does not include the pr and the provider included, direct decoding by the provider could cause issues.

2、If the consumer includes the pr and the provider does not include the pr, the provider will receive the encoded value without decoding it. If the consumer attachment includes special values (e.g. '%' or '+'), they will be encoded, resulting in unexpected behavior.

To ensure compatibility between different versions, it may be necessary to add a configuration option in the URL registered by the provider in registration center. This configuration can control whether the consumer-side needs to encoding attachment and provider needs to decoding